comp/core/profiler/impl/profiler.go

```diff
@@ -0,0 +1,333 @@
+					name: service + "-1st-heap.pprof",
+					path: "/heap",
+				},
+				{
+					// CPU profile
+					name: service + "-cpu.pprof",
+					path: fmt.Sprintf("/profile?seconds=%d", seconds),
+				},
+				{
+					// 2nd heap profile
+					name: service + "-2nd-heap.pprof",
+					path: "/heap",
+				},
+				{
+					// mutex profile
+					name: service + "-mutex.pprof",
+					path: "/mutex",
+				},
+				{
+					// goroutine blocking profile
+					name: service + "-block.pprof",
+					path: "/block",
+				},
+				{
+					// Trace
+					name: service + ".trace",
+					path: fmt.Sprintf("/trace?seconds=%d", seconds),
+				},
+			} {
+				b, err := get(prof.path)
+				if err != nil {
+					return err
+				}
+				pdata[prof.name] = b
+			}
+			return nil
+		}
+	}
+
+	agentCollectors := map[string]agentProfileCollector{}
+	if p.coreAgentEnabled() {
+		agentCollectors["core"] = serviceProfileCollector(tcpGet("expvar_port", false), seconds)
+	}
+
+	if p.securityAgentEnabled() {
+		agentCollectors["security-agent"] = serviceProfileCollector(tcpGet("security_agent.expvar_port", false), seconds)
+	}
+
+	if p.processAgentEnabled() {
+		agentCollectors["process"] = serviceProfileCollector(tcpGet("process_config.expvar_port", false), seconds)
+	}
+
+	if p.apmEnabled() {
+		traceCpusec := p.apmTraceSeconds(seconds)
+		agentCollectors["trace"] = serviceProfileCollector(tcpGet("apm_config.debug.port", true), traceCpusec)
+	}
+
+	if p.sysProbeEnabled() {
+		client := &http.Client{
+			Transport: &http.Transport{
+				DialContext: sysprobeclient.DialContextFunc(p.sysProbeCfg.GetString("system_probe_config.sysprobe_socket")),
+			},
+		}
+
+		sysProbeGet := func() pprofGetter {
+			return func(path string) ([]byte, error) {
+				var buf bytes.Buffer
+				pprofURL := sysprobeclient.DebugURL("/pprof" + path)
+				req, err := http.NewRequest(http.MethodGet, pprofURL, &buf)
+				if err != nil {
+					return nil, err
+				}
+
+				res, err := client.Do(req)
+				if err != nil {
+					return nil, err
+				}
+				defer res.Body.Close()
+
+				return io.ReadAll(res.Body)
+			}
+		}
+
+		agentCollectors["system-probe"] = serviceProfileCollector(sysProbeGet(), seconds)
+	}
+
+	var errs error
+	for name, callback := range agentCollectors {
+		if err := callback(name); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error collecting %s agent profile: %v", name, err))
+		}
+	}
+
+	return pdata, errs
+}
+
+func (p profiler) setProfilerSetting(settingName string, newValue int, fb flaretypes.FlareBuilder) func() {
+	oldValue, err := p.settingsComponent.GetRuntimeSetting(settingName)
+	if err != nil {
+		_ = fb.Logf("Unable to access %s setting, ignoring value provided by flare args: %v", settingName, err)
+
+		return nil
+	} else if newValue <= 0 || newValue == oldValue {
+		return nil
+	}
+
+	err = p.settingsComponent.SetRuntimeSetting(settingName, newValue, model.SourceAgentRuntime)
+	if err != nil {
+		_ = fb.Logf("Unable to set the %s: %v", settingName, err)
```

Comment:
If we fail set the new value should we retun nil? It seems that we are returning a defer  block that sets the settings to the old value, but we never set it to the new value 😄